indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -451,29 +451,51 @@ def _process_args(self):
         self.export_config = self.list_generator.get_list_export_config()
 
 
-class RHRegistrationsExportPDFTable(RHRegistrationsExportBase):
-    """Export registration list to a PDF in table style."""
+class RHRegistrationsExportPDF(RHRegistrationsExportBase):
+    """Export registration list to a PDF in table or book style."""
 
-    def _process(self):
-        pdf = RegistrantsListToPDF(self.event, reglist=self.registrations, display=self.export_config['regform_items'],
-                                   static_items=self.export_config['static_item_ids'])
-        try:
-            data = pdf.getPDFBin()
-        except Exception:
-            if config.DEBUG:
-                raise
-            raise NoReportError(_('Text too large to generate a PDF with table style. '
-                                  'Please try again generating with book style.'))
-        return send_file('RegistrantsList.pdf', BytesIO(data), 'application/pdf')
+    _base_spec = getattr(RHRegistrationsExportBase, 'normalize_url_spec', RH.normalize_url_spec)
 
+    normalize_url_spec = {
+        **_base_spec,
+        'preserved_args': (_base_spec.get('preserved_args', set()) | {'export_type'})
+    }
 
-class RHRegistrationsExportPDFBook(RHRegistrationsExportBase):
-    """Export registration list to a PDF in book style."""
+    def _process_args(self):
+        super()._process_args()
+        self.export_type = request.view_args.get('export_type')
```

Comment:
- Use webargs `@use_kwargs()` to get the export type.
- Do not use `.get()` on dicts when you know the value always exists (or expect it to always exist)